tests/functional/u/unspecified-encoding_py3.py

```diff
@@ -0,0 +1,7 @@
+"""Warnings for using open() without specifying an encoding"""
+# pylint: disable=consider-using-with
+
+NAME = "foo.bar"
+open(NAME, "wb", encoding="utf-8")
+open(NAME, "w", encoding=None)  # [unspecified-encoding]
+open(NAME, "rb")  # [unspecified-encoding]
```

Comment:
Let's add the example given in the issue :)
```suggestion
open(NAME, "rb")  # [unspecified-encoding]

with open(filename) as f:  # [unspecified-encoding]
    ...

with open(filename, encoding=None) as f:  # [unspecified-encoding]
    ...


with open(filename, encoding="utf8", errors="surrogateescape") as f:
    ...

locale_encoding = getattr(io, "LOCALE_ENCODING", None)
with open(filename, encoding=locale_encoding) as f:
    ...
```
Also we should probably check more use case to handle other open method (method named open, other method from io, etc.)